cmd/nknc/commands/asset.go

```diff
@@ -0,0 +1,213 @@
+
+	assetCmd.Flags().MarkHidden("password")
+}
+
+func parseAddress() common.Uint160 {
+	if address := to; address != "" {
+		pg, err := common.ToScriptHash(address)
+		if err != nil {
+			fmt.Println("invalid receiver address")
+			os.Exit(1)
+		}
+		return pg
+	}
+	fmt.Println("missing flag [--to]")
+	os.Exit(1)
+	return common.EmptyUint160
+}
+
+func assetAction() error {
+	var err error
+	var txnFee common.Fixed64
+
+	if Fee == "" {
+		txnFee = common.Fixed64(0)
+	} else {
+		txnFee, err = common.StringToFixed64(Fee)
+		if err != nil {
+			fmt.Fprintln(os.Stderr, err)
+			return err
+		}
+	}
+
+	var resp []byte
+	switch {
+	case issue:
+		if WalletFile == "" {
+			fmt.Println("wallet file is required with [--wallet]")
+			return nil
+		}
+
+		myWallet, err := vault.OpenWallet(WalletFile, getPassword(WalletPassword))
+		if err != nil {
+			fmt.Fprintln(os.Stderr, err)
+			os.Exit(1)
+		}
+
+		if assetname == "" {
+			fmt.Println("asset name is required with [--name]")
+			return nil
+		}
+
+		if symbol == "" {
+			fmt.Println("asset symbol is required with [--symbol]")
+			return nil
+		}
+
+		totalSupply, err := common.StringToFixed64(value)
+		if err != nil {
+			fmt.Fprintln(os.Stderr, err)
+			return err
+		}
+
+		if precision > config.MaxAssetPrecision {
+			err := fmt.Errorf("precision is larger than %v", config.MaxAssetPrecision)
+			fmt.Fprintln(os.Stderr, err)
+			return err
+		}
+
+		txn, err := api.MakeIssueAssetTransaction(myWallet, assetname, symbol, totalSupply, precision, Nonce, txnFee)
+		if err != nil {
+			fmt.Fprintln(os.Stderr, err)
+			return err
+		}
+
+		buff, err := txn.Marshal()
+		if err != nil {
+			fmt.Fprintln(os.Stderr, err)
+			return err
+		}
+
+		resp, err = client.Call(Address(), "sendrawtransaction", 0, map[string]interface{}{"tx": hex.EncodeToString(buff)})
+		if err != nil {
+			fmt.Fprintln(os.Stderr, err)
+			return err
+		}
+	case transfer:
+		myWallet, err := vault.OpenWallet(WalletFile, getPassword(WalletPassword))
+		if err != nil {
+			fmt.Fprintln(os.Stderr, err)
+			os.Exit(1)
+		}
+		receipt := parseAddress()
+		amount, err := common.StringToFixed64(value)
+		if err != nil {
+			fmt.Fprintln(os.Stderr, err)
+			return err
+		}
+
+		if Nonce == 0 {
+			remoteNonce, _, err := client.GetNonceByAddr(Address(), myWallet.Address, true)
+			if err != nil {
+				return err
+			}
+			Nonce = remoteNonce
+		}
+
+		txn, err := api.MakeTransferTransaction(myWallet, receipt, Nonce, amount, txnFee)
+		if err != nil {
+			fmt.Fprintln(os.Stderr, err)
+			return err
+		}
+
+		buff, err := txn.Marshal()
+		if err != nil {
+			fmt.Fprintln(os.Stderr, err)
+			return err
+		}
+
+		resp, err = client.Call(Address(), "sendrawtransaction", 0, map[string]interface{}{"tx": hex.EncodeToString(buff)})
+		if err != nil {
+			fmt.Fprintln(os.Stderr, err)
+			return err
+		}
+	default:
+	}
+
+	FormatOutput(resp)
+
+	return nil
+}
+
+func getPassword(passwd string) []byte {
```

Comment:
The same function is duplicated in `root.go`

Reply:
thanks.